extension/intents/tabs/tabs.js

```diff
@@ -469,13 +469,58 @@ async function getMatchingTabs(options) {
   let matchingTabs = await browser.tabs.query(tabQuery);
 
   if (options.query !== undefined) {
-    matchingTabs = matchingTabs.filter(tab => {
-      const query = options.query.toLowerCase();
-      return (
-        new URL(tab.url).origin.includes(query) ||
-        tab.title.toLowerCase().includes(query)
-      );
-    });
+    const fuseOptions = {
+      id: "tabId",
+      shouldSort: true,
+      tokenize: true,
+      findAllMatches: true,
+      includeScore: true,
+      threshold: 0.3,
+      location: 0,
+      distance: 100,
+      maxPatternLength: 32,
+      minMatchCharLength: 3,
+      keys: [
+        {
+          name: "title",
+          weight: 0.8,
+        },
+        {
+          name: "url",
+          weight: 0.2,
+        },
+      ],
+    };
+
+    let combinedTabContent = [];
+
+    for (const tab of matchingTabs) {
+      const result = {
+        tabId: tab.id,
+        title: tab.title,
+        url: tab.url,
+      };
+
+      combinedTabContent.push(result);
+    }
+
+    combinedTabContent = combinedTabContent.flat();
```

Comment:
This probably isn't necessary? It looks like the array should be flat already.

Reply:
@ianb Thanks for the suggestion, I have removed the unnecessary code.